packages/core/src/animation/animationCurve/AnimationCurve.ts

```diff
@@ -51,12 +51,18 @@ export abstract class AnimationCurve<V extends KeyframeValueType> {
    */
   addKey(key: Keyframe<V>): void {
     const { time } = key;
-    this.keys.push(key);
-    if (time > this._length) {
+    const { keys } = this;
+
+    if (time >= this._length) {
+      keys.push(key);
       this._length = time;
+    } else {
+      let index = keys.length;
+      while (index > 0 && key.time < keys[index - 1].time) {
```

Comment:
while(--index > 0) {
   if( time < keys[index]) {
       keys.splice(index, 0, key);
       break;
   }
}

Reply:
![image](https://user-images.githubusercontent.com/4967545/209286215-e84437b9-4026-464d-ade4-944ef11a0e24.png)
This opt has bug